### PR TITLE
Add docs for withName() command decorator

### PR DESCRIPTION
**Pull request by prateekma**

### CODE REVIEW

**Comment by Daltz333** [comment]
What's the point of a command having a name? The intended usage?

**Comment by prateekma** [comment]
The name shows up when you send a command to a dashboard (via the Sendable interface).

**Comment by Daltz333** [comment]
Can this information be added?

File: source/docs/software/commandbased/convenience-features.rst [hunk 1]
```diff
@@ -453,6 +453,20 @@ The ``deadlineWith()`` `decorator <https://first.wpi.edu/FRC/roborio/release/doc
    // Will be a parallel deadline group containing fooCommand, barCommand, and bazCommand; fooCommand is the deadline
    button.whenPressed(fooCommand.deadlineWith(barCommand, bazCommand));
 
+withName (Java only)
+^^^^^^^^^^^^^^^^^^^^
+
+.. note:: This decorator is not supported in C++ due to technical constraints - users should set the name of the command inside their command class instead.
+
+.. todo:: Add link to release documentation when release is made.
```
Comment (sciencewhiz) [inline source/docs/software/commandbased/convenience-features.rst:461]: Is it a different link then this? https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj2/command/CommandBase.html#withName(java.lang.String)

Comment (prateekma) [reply source/docs/software/commandbased/convenience-features.rst:461]: No, that is correct.

**Review by calcmogul — APPROVED** [review]

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**